torch/csrc/jit/codegen/cuda/lower_unroll.cpp

```diff
@@ -84,8 +85,33 @@ void UnrollPass::handle(kir::Expr* expr) {
     const auto thread_pred = isReductionInitExpr(expr)
         ? ir_builder.create<kir::Bool>(true)
         : getThreadPredicate(out_tv);
-    const auto pred =
-        PredicateCompute::getInlinePredicate(expr, for_loops_, thread_pred);
+
+    // Vectorized expressions should never use inline predicates
+    kir::Bool* vectorized_pred = nullptr;
+    if (std::any_of(
+            for_loops_.begin(), for_loops_.end(), [](const kir::ForLoop* fl) {
+              return fl->iter_domain()->parallelType() ==
+                  ParallelType::Vectorize;
+            })) {
```

Comment:
Can this be true? If I understand the logic in `handle(ForLoop*)`, if a vectorized `ForLoop` is found, `handle(Expr*` doesn't seem to be called for those inside the vectorized loop.

Reply:
It can be true in the else statement of an unrolled loop

Reply:
Thanks. Makes sense.